main.go

```diff
@@ -14,91 +14,268 @@
+
+	// The number of errors seen
+	Errors uint64
+	// The last error seen
+	LastError error
+}
+
+func printDataOut(ctx context.Context, cancel context.CancelFunc) {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+	go func() {
+		<-sigc
+		httpListener.Close()
+		cancel()
+	}()
+
+	lastPrint := false
+	for !lastPrint {
+		time.Sleep(1 * time.Second)
+
+		// We wait for requests and clients to exit so that
+		// the request counters are as accurate as possible.
+		// lastPrint = activeClients.Load() == 0 && activeRequests.Load() == 0 && ctx.Err() != nil
+		if activeClients.Load() == 0 && activeRequests.Load() == 0 {
+			if ctx.Err() != nil {
+				lastPrint = true
+			}
+		}
+
+		if time.Since(start).Seconds() > float64(seconds) {
+			httpListener.Close()
+			cancel()
+		}
+
+		list := generateDataPoint()
+
+		if printJSON {
+			for _, v := range list {
+				outb, _ := json.Marshal(v)
+				fmt.Println(string(outb))
+			}
+		} else {
+			printToConsole(list)
+		}
+
 	}
-	return p
-}()
+}
 
-var selfDetectPort = func() string {
-	if sp := os.Getenv("HPERF_SELF_PORT"); sp != "" {
-		return sp
+func generateDataPoint() (list []dataPoint) {
+	list = make([]dataPoint, 0, len(Readers))
+	for ri, rv := range Readers {
+		if rv == nil {
+			continue
+		}
+		for wi, wv := range Writers {
+			if wv == nil || wv.Host != rv.Host {
+				continue
+			}
+
+			w := Writers[wi]
+			r := Readers[ri]
+
+			list = append(list, dataPoint{
+				Created:           time.Now(),
+				RX:                w.RX.Swap(0),
+				RXCount:           w.count.Load(),
+				Local:             currentHost,
+				RealRemote:        w.RealHost,
+				TTFBLow:           r.TTFBLow.Swap(999999),
+				TTFBHigh:          r.TTFBHigh.Swap(0),
+				LowestTransferMS:  r.LowestTransferMS.Swap(999999),
+				HighestTransferMS: r.HighestTransferMS.Swap(0),
+				TX:                r.TX.Swap(0),
+				TXCount:           r.count.Load(),
+				Errors:            r.Errors.Swap(0),
+				Remote:            r.Host,
+				LastError:         globalErr,
+			})
+		}
 	}
-	sp, err := strconv.Atoi(port)
-	if err != nil {
-		log.Fatal(err)
+	return
+}
+
+func (d dataPoint) String() string {
+	le := ""
+	if d.LastError != nil {
+		le = "|| " + d.LastError.Error()
+	}
+	realRemote := ""
+	if d.RealRemote != d.Remote {
+		realRemote = "(" + d.RealRemote + ")"
 	}
-	sp++
-	return strconv.Itoa(sp)
-}()
 
-var uniqueStr = uuid.New().String()
+	if latency {
+		return fmt.Sprintf(
+			"%s - %s%s || MS %d>%d || E %d %s",
+			d.Local,
+			d.Remote,
+			realRemote,
+			d.HighestTransferMS,
+			d.LowestTransferMS,
+			d.Errors,
+			le,
+		)
+	}
 
-var (
-	dataIn  uint64
-	dataOut uint64
-)
+	if stream {
+		return fmt.Sprintf(
+			"%s - %s%s || RX (%d) %s/s || TX (%d) %s/s || E %d %s",
+			d.Local,
+			d.Remote,
+			realRemote,
+			d.RXCount,
+			humanize.Bytes(d.RX),
+			d.TXCount,
+			humanize.Bytes(d.TX),
+			d.Errors,
+			le,
+		)
+	}
```

Comment:
use our `pkg/console.NewTable()` to format this properly or use tablewriter - look for the console.NewTable() usage in `mc admin service restart alias/` UI